src/rest_api/classes/interaction/core.clj

```diff
@@ -45,16 +46,56 @@
   (some-fn :molecule/id :gene/id :rearrangement/id :feature/id))
 
 (def int-rules
-  '[[(gene-interaction ?gene ?int)
-     [?ih :interaction.interactor-overlapping-gene/gene ?gene]
-     [?int :interaction/interactor-overlapping-gene ?ih]]
-    [(gene-neighbour ?gene ?neighbour)
-     (gene-interaction ?gene ?ix)
-     [?ix :interaction/interactor-overlapping-gene ?ih]
+  '[[(gene->interaction-3 ?gene ?h ?int)
+     [?h :interaction.interactor-overlapping-gene/gene ?gene]
+     [?int :interaction/interactor-overlapping-gene ?h]]
+    [(feature->interaction-3 ?feature ?h ?int)
+     [?h :interaction.feature-interactor/feature ?feature]
+     [?int :interaction/feature-interactor ?h]]
+    [(molecule->interaction-3 ?molecule ?h ?int)
+     [?h :interaction.molecule-interactor/molecule ?molecule]
+     [?int :interaction/molecule-interactor ?h]]
+    [(rearrangement->interaction-3 ?rearrangement ?h ?int)
+     [?h :interaction.rearrangement/rearrangement ?rearrangement]
+     [?int :interaction/rearrangement ?h]]
+    [(x->interaction-3 ?x ?h ?ix)
+     (or (gene->interaction-3 ?x ?h ?ix)
```

Comment:
a few questions and comments...

1.  New things I see are the introduction of "type" (e.g molecule, feature) specific interaction relations - were these being found before and are now just being literally expanded for readability?
 
2. I'm not entirely sure the naming of rule-heads (having the number-of-params as a suffix as a naming thing)

3.  _seems_ counter intuitive to have an `(or` statement  (since rules of the same name naturally *or* together; but this is a very complex area, so can understand why it's easier to break the  rules up. Related to `1.` am sure.
Perhaps naming the rules individually with defs and making `int-roles` a composite of them would help?
In that way one could still keep the same "rule head" names (would perhaps remove the need to